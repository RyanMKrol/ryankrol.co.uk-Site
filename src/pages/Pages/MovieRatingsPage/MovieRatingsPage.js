import React, { Component } from 'react';

import { DataCollection, MovieRatingsItem } from '../../../components';

import styles from './MovieRatingsPage.module.css';

import { baseEndpoint, generateCustomSorter } from '../../../utils/ProdUtils';

/**
 * Page for rendering sub-section of Ratings content
 */
class MovieRatingsPage extends Component {
  /**
   * @class
   */
  constructor() {
    super();

    this.endpoint = `${baseEndpoint()}/ratings/movie`;
  }

  /**
   * Render method
   *
   * @returns {string} JSX Content
   */
  render() {
    return (
      <div className={styles.container}>
        <h2>Movies I've Watched</h2>
        <br />
        <DataCollection
          endpoint={this.endpoint}
          itemTag={MovieRatingsItem}
          sorter={generateCustomSorter('blind')}
        />
      </div>
    );
  }
}

export default MovieRatingsPage;
