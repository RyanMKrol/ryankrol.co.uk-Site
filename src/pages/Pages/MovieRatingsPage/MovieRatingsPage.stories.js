import React from 'react';

import MovieRatingsPage from './MovieRatingsPage';

export default {
  title: 'Pages/MovieRatingsPage',
};

/**
 * default story
 *
 * @returns {React.Component} JSX for the MovieRatingsPage
 */
export const normal = () => <MovieRatingsPage />;
