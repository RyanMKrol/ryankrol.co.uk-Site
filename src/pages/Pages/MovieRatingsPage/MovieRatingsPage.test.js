import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import MovieRatingsPage from './MovieRatingsPage';

it('renders MovieRatingsPage correctly', () => {
  const header = shallow(<MovieRatingsPage />);
  expect(toJson(header)).toMatchSnapshot();
});
