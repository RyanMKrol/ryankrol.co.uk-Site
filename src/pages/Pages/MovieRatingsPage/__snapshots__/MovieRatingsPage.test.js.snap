// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MovieRatingsPage correctly 1`] = `
<div
  className="container"
>
  <h2>
    Movies I've watched
  </h2>
  <br />
  <DataCollection
    endpoint="http://www.ryankrol.co.uk/api/ratings/movie"
    itemTag={[Function]}
    sorter={[Function]}
  />
</div>
`;
